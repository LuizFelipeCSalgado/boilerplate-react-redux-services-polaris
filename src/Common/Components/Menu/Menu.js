import React from 'react';
import { NavLink } from 'react-router-dom';

import PropTypes from 'prop-types';

import './Menu.scss';

function Menu({ items }) {
  return (
    <div className="menu">
      {
        items
          ? items.map(item => (
            <div key={`menu-item-${item.id}`}>
              <NavLink to={item.path}> {item.name} </NavLink>
            </div>
          ))
          : null
      }
    </div>
  )
}

Menu.propTypes = {
  items: PropTypes.array.isRequired,
}

export default Menu;