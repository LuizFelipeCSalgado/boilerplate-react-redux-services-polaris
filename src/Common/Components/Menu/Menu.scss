@import 'Styles/variables.scss';

.menu {
  display: flex;
  flex-direction: column;
  box-shadow: inset -1px 0 0 0 #DFE3E8;
  overflow-x: hidden;
  height: 100%;
}