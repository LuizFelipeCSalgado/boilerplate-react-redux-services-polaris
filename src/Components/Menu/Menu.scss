@import 'Common/Assets/Styles/variables.scss';

.menu {
  display: flex;
  flex-direction: column;
  box-shadow: inset -1px 0 0 0 #dfe3e8;
  overflow-x: hidden;
  height: 100%;
}
