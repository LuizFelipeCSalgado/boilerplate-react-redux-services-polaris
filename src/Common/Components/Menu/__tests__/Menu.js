import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import Menu from '../Menu';

describe('Menu', () => {
  let wrapper = null;
  const props = {
    items: [
      { id: "home", isMenu: true, name: "Home", path: "/home" },
      { id: "listagem", isMenu: true, name: "Listagem", path: "/listagem" }
    ]
  };

  const mountComponent = (newProps = {}) => {
    wrapper = mount(
      <Router>
        <Menu {...props} {...newProps} />
      </Router>
    );
  };

  beforeEach(() => {
    mountComponent();
  });

  it('should render the without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

})
