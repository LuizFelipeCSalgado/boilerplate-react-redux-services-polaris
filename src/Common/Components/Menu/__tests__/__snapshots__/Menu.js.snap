// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render the without crashing 1`] = `null`;
